File: bl-central-server/bl-sensor-data-collector/src/main/java/org/jesperancinha/logistics/sensor/collector/converter/TrainConverter.java
package org.jesperancinha.logistics.sensor.collector.converter;

import org.jesperancinha.logistics.jpa.model.Train;
import org.jesperancinha.logistics.jpa.model.TrainLog;
import org.jesperancinha.logistics.sensor.collector.data.TrainLogDto;

public class TrainConverter {
    public static TrainLog toModel(TrainLogDto trainLogDto, Train train) {
        return TrainLog.builder()
            .checkInOut(trainLogDto.type())
            .timestamp(trainLogDto.timestamp())
            .train(train)
            .lat(trainLogDto.lat())
            .lon(trainLogDto.lon())
            .build();
    }
}

File: bl-central-server/bl-sensor-data-collector/src/main/kotlin/org/jesperancinha/logistics/sensor/collector/converter/TrainConverter.kt
package org.jesperancinha.logistics.sensor.collector.converter

import org.jesperancinha.logistics.jpa.dao.Train
import org.jesperancinha.logistics.jpa.dao.TrainLog
import org.jesperancinha.logistics.sensor.collector.data.TrainLogDto

object TrainConverter {
    fun toModel(trainLogDto: TrainLogDto, train: Train?): TrainLog {
        return TrainLog(
            checkInOut = trainLogDto.type,
            timestamp = trainLogDto.timestamp,
            train = train,
            lat = trainLogDto.lat,
            lon = trainLogDto.lon,
            weight = trainLogDto.weight,
            carriageId = trainLogDto.carriageId
        )
    }
}